kmapper/cover.py

```diff
@@ -94,3 +93,104 @@ def find_entries(self, data, cube, verbose=0):
         hypercube = data[np.invert(np.any(entries == False, axis=1))]
 
         return hypercube
+
+
+class CoverBounds(Cover):
+    """
+    Class that allows cover to be defined using upper and lower bounds for given dimensions.
+    """
+
+    def __init__(self,
+                 n_cubes=10,
+                 perc_overlap=0.2,
+                 # Deprecated parameters:
+                 nr_cubes=None,
+                 overlap_perc=None,
+                 # Bounds parameters
+                 limits=None):
+        """
+        limits: Numpy Array (n_dim,2)
+            (lower bound, upper bound) for every dimension
+            If a value is set to np.float('inf'), the bound will be assumed to be the min/max value of the dimension
+            Instead, if limits == None, this class works identically to the original Cover class.
+            i.e.
+                [[min_1, max_1],
+                 [min_2, max_2],
+                 [min_3, max_3]]
+        """
+        Cover.__init__(self, n_cubes, perc_overlap, nr_cubes, overlap_perc)
+        self.limits = limits
+
+        # Check limits can actually be handled and are set appropriately
+        NoneType = type(None)
```

Comment:
defining this NoneType variable is weird. Why do it?  Please just use type(None) when you need it.